lib/client.js

```diff
@@ -303,10 +303,15 @@ Client.prototype.sendGroupRequest = function (encode, decode, requestArgs) {
 
   var request = encode.apply(null, requestArgs);
   var broker = this.brokerForLeader(this.coordinatorId);
-
-  if (!broker || !broker.socket || broker.socket.error || broker.socket.destroyed) {
+  var brokerError = null;
+  if (!broker || !broker.socket) {
```

Comment:
Do we need this check since the socket should always exist on a broker if it's available?